docs/products/kafka/kafka-connect/howto/azure-blob-source.md

```diff
@@ -0,0 +1,185 @@
+---
+title: Create an Azure Blob Storage source connector for Aiven for Apache Kafka®
+sidebar_label: Azure Blob source connector
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+import ConsoleLabel from "@site/src/components/ConsoleIcons";
+
+The Azure Blob Storage source connector allows you to ingest data from Azure Blob Storage containers into Apache Kafka® topics for real-time processing, analytics, or disaster recovery.
+
+## Prerequisites
+
+- An [Aiven for Apache Kafka® service](/docs/products/kafka/kafka-connect/howto/enable-connect)
+  with Apache Kafka Connect enabled, or a
+  [dedicated Aiven for Apache Kafka Connect® service](/docs/products/kafka/kafka-connect/get-started#apache_kafka_connect_dedicated_cluster).
+- An Azure Storage container with data to stream into an Apache Kafka topic.
+- Authentication credentials using one of the following methods:
+  - Shared Access Signature (SAS) token: `azblob.sas.token`
+  - Storage account key: `azure.storage.connection.string`
+
+  :::note
+  Use either `azblob.sas.token` or `azure.storage.connection.string`, not both. If both
+  are provided, the SAS token is used.
+  :::
+
+- Azure Blob Storage details:
+  - `azure.storage.container.name`: Name of the container to read from.
+  - Optional: `azure.blob.prefix` or `file.name.prefix` to filter files.
+- A target Apache Kafka topic that exists before you create the connector.
+
+## Supported input formats
+
+Specify the input format of your files using the `format.class` parameter:
+
+| Format    | Description                                | Configuration               |
+| --------- | ------------------------------------------ | --------------------------- |
+| `csv`     | Flat structure with comma-separated values | `format.class=csv`     |
+| `jsonl`   | One JSON object per line                   | `format.class=jsonl`   |
+| `json`    | JSON array of objects                      | `format.class=json`    |
+| `avro`    | Avro Object Container File                 | `format.class=avro`    |
+| `parquet` | Apache Parquet columnar format             | `format.class=parquet` |
+
+## Create an Azure Blob Storage source connector configuration file
+
+Create a file named `azure_blob_source_config.json` with the following configuration:
+
+```json
+{
+  "name": "azure-blob-source",
+  "connector.class": "io.aiven.connect.azure.blob.storage.source.AzureBlobStorageSourceConnector",
+  "azure.storage.connection.string": "your-connection-string",
+  "azure.storage.container.name": "your-container",
+  "azure.blob.prefix": "data/logs/",
+  "format.class": "json",
+  "kafka.topic": "blob-ingest-topic",
+  "format.output.fields": "key,value,offset,timestamp",
+  "file.name.prefix": "data/logs/",
+  "file.compression.type": "gzip",
+  "tasks.max": 1,
+  "poll.interval.ms": 10000
+}
```

Comment:
The configuration property `file.name.template` is mandatory. If not set no objects will be processed and the connector task will fail with error:
```
org.apache.kafka.common.config.ConfigException: Source name format is missing please configure the expected source to include the partition pattern.
```

An example of valid configuration is:
```
  "file.name.template": "{{topic}}-{{timestamp}}.gz",
```
Other format placeholders are documented here: https://github.com/Aiven-Open/cloud-storage-connectors-for-apache-kafka/blob/main/s3-source-connector/README.md#s3-object-key-name-format